File: OneCodeTeam/Load-all-Toolbox-items-in-an-Assembly-in-15865/[VB]-Load all Toolbox items in an Assembly in VSPackage (VBVSPackageToolbox)/VB/VBVSPackageToolbox/MyControl.vb
'**************************** Module Header ******************************'
' Module Name:  MyControl.vb
' Project:      VBVSPackageToolbox
' Copyright (c) Microsoft Corporation.
' 
' This UserControl will be displayed as the content of the ToolWindow.
' 
' This source is subject to the Microsoft Public License.
' See http://www.microsoft.com/en-us/openness/resources/licenses.aspx#MPL.
' All other rights reserved.
' 
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
' WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
'**************************************************************************'

Imports System.Windows.Forms
Imports System.Drawing
Imports System.ComponentModel

<DisplayName("ToolboxMemberDemo"), _
Description("Custom toolbox item from package LoadToolboxMembers."), _
ToolboxItem(True), _
ToolboxBitmap(GetType(MyControl), "MyControl.bmp")> _
Public Class MyControl

    Private Sub btnBrowse_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnBrowse.Click
        Dim opg As New OpenFileDialog()
        opg.Multiselect = False
        If opg.ShowDialog() = DialogResult.OK Then
            Me.tbFileName.Text = opg.FileName
        End If
    End Sub
End Class

File: OneCodeTeam/Load-all-Toolbox-items-in-an-Assembly-in-233/[C#]-Load all Toolbox items in an Assembly in VSPackage (CSVSPackageToolbox)/C#/CSVSPackageToolbox/MyControl.cs
/****************************** Module Header ******************************\
 Module Name:  MyControl.cs
 Project:      CSVSPackageToolbox
 Copyright (c) Microsoft Corporation.
 
 This UserControl will be displayed as the content of the ToolWindow.
 
 This source is subject to the Microsoft Public License.
 See http://www.microsoft.com/en-us/openness/resources/licenses.aspx#MPL.
 All other rights reserved.
 
 THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
 EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
 WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
\***************************************************************************/

using System;
using System.ComponentModel;
using System.Drawing;
using System.Windows.Forms;

namespace MyCompany.CSVSToolbox
{
    // Set the display name and custom bitmap to use for this item.
    // The build action for the bitmap must be "Embedded Resource".
    [DisplayName("ToolboxMemberDemo")]
    [Description("Custom toolbox item from package LoadToolboxMembers.")]
    [ToolboxItem(true)]
    [ToolboxBitmap(typeof(MyControl), "MyControl.bmp")]
    public partial class MyControl : UserControl
    {
        public MyControl()
        {
            InitializeComponent();
        }

        private void btnBrowse_Click(object sender, EventArgs e)
        {
            OpenFileDialog opg = new OpenFileDialog();
            opg.Multiselect = false;
            if (opg.ShowDialog() == DialogResult.OK)
                this.tbFileName.Text = opg.FileName;
        }
    }
}
